### ISSUE TITLE
vue/valid-define-emits and -props treat types as local variable and fail

### ISSUE THREAD

**Issue by larvanitis** [vuejs/eslint-plugin-vue#1656]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version: 7.32.0** 
- **eslint-plugin-vue version: 7.19.1** 
- **Node version: 14.17.4** 
- **Operating System: Arch Linux** 

**Please show your full configuration:**
```js
module.exports = {
  root: true,

  parserOptions: {
    extraFileExtensions: ['.vue'],
    parser: '@typescript-eslint/parser',
    project: './tsconfig.json',
    tsconfigRootDir: __dirname,
    ecmaVersion: 2018,
    sourceType: 'module',
  },

  env: {
    browser: true,
  },

  extends: [
    'plugin:vue/vue3-essential',
  ],

  plugins: [
    '@typescript-eslint',
    'vue',
  ],

  rules: {
    'vue/valid-define-emits': 'error',
    'vue/valid-define-props': 'error',
  },
};
```

**What did you do?**	
```vue
<script setup lang="ts">
import type { PropType } from 'vue';

const props = defineProps({
  myProp: Array as PropType<string[]>,
});             // ^ error


type X = string;

const emit = defineEmits({
  myEvent: (x: X) => true,
});         // ^ error
</script>
```

**What did you expect to happen?**
Use types without error.

**What actually happened?**
```
/path/to/Component.vue
   7:20  error  `defineProps` are referencing locally declared variables  vue/valid-define-props
  11:15  error  `defineEmits` are referencing locally declared variables  vue/valid-define-emits
```
